karate-tests/src/test/java/patients/patientAccess/getCoverageDetails.feature

```diff
@@ -0,0 +1,78 @@
+@no-oas
+Feature: Patient Access (Retrieve)
+    Retrieve a chargeable snippet for a patient
+
+Background:
+    * def utils = call read('classpath:helpers/utils.feature')
+    * json patientCoverageResultEntry  = karate.readAsString('classpath:schemas/searchSchemas/patientCoverageResultEntry.json')
+    * json coverageBundle = karate.readAsString('classpath:schemas/searchSchemas/patientCoverageBundle.json')
+    
+    * configure url = baseURL
+
+ @sandbox
+   Scenario: Happy path - Retrieve patient coverage details
+    * def p9number = '9733162868'
+    * def accessToken = karate.call('classpath:auth/auth-redirect.feature', {userID: p9number, scope: 'nhs-login'}).accessToken
+    * def requestHeaders = call read('classpath:auth/auth-headers.js')
+    * configure headers = requestHeaders
+    * path 'Coverage'
+    * param beneficiary:identifier = p9number
+    * method get
+    * status 200
+    * match response == coverageBundle
+    * match response.entry[0].resource.status == 'active'
+    * match response.entry[0].resource.beneficiary.identifier.value == p9number
+    * match response.entry[0].resource.identifier[0].assigner contains { display: '#notnull'}
+
+  @sandbox
+ Scenario: Happy path - patient has no coverage details
+    * def P9WithNoCoverage = '9733162876'
+    * def accessToken = karate.call('classpath:auth/auth-redirect.feature', {userID: P9WithNoCoverage, scope: 'nhs-login'}).accessToken
+    * def requestHeaders = call read('classpath:auth/auth-headers.js')
+    * configure headers = requestHeaders
+    * path 'Coverage'
+    * param beneficiary:identifier = P9WithNoCoverage
+    * method get
+    * status 200
+    * match response.entry[0] == '#notpresent'
+
+ # 9732019735 is displayed in retained record 9732019638
+Scenario: Retrieve patient current coverage details when superseded NHS number is sent
+    * def mergedP9number = '9732019735'
+    * def retainedRecord = '9732019638' 
+    * def ehicCardNo = '12346 00002 02 0002'
+    * def accessToken = karate.call('classpath:auth/auth-redirect.feature', {userID: mergedP9number, scope: 'nhs-login'}).accessToken
+    * def requestHeaders = call read('classpath:auth/auth-headers.js')
+    * configure headers = requestHeaders
+    * path 'Coverage'
+    * param beneficiary:identifier = mergedP9number
+    * method get
+    * status 200
+    * match response == coverageBundle
+    * match response.entry[0].resource.status == 'active'
+    * match response.entry[0].resource.beneficiary.identifier.value != mergedP9number 
```

Comment:
Feel free to resolve if you disagree, but I would argue we don't need this check as we already prove that the returned number != mergedP9number, by confirming that returned number == retainedRecord on the following line

Reply:
not sure, but removed the  returned number == retainedRecord  check

Reply:
If we're just keeping one it should be the line
`match response.entry[0].resource.beneficiary.identifier.value == retainedRecord`

That way we prove that we're getting back the nhs number that we expect, and because we already know that retainedRecord != mergedP9number, we don't need to check mergedP9number against the response, so the line
`match response.entry[0].resource.beneficiary.identifier.value != mergedP9number`
can be removed.

Reply:
Added `match response.entry[0].resource.beneficiary.identifier.value == retainedRecord` line back